packages/paste-website/src/component-examples/ListboxPrimitiveExamples.ts

```diff
@@ -0,0 +1,250 @@
+export const defaultExample = `
+const DefaultExample = () => {
+  const [items] = React.useState(['Item 1', 'Item 2', 'Item 3']);
+  const [selected, setSelected] = React.useState();
+  const listbox = useListboxPrimitiveState();
+  return (
+    <ListboxPrimitive {...listbox} aria-label="My listbox">
+      <Stack orientation="vertical" spacing="space0">
+        {items.map((item) => (
+          <ListboxPrimitiveItem
+            key={item}
+            {...listbox}
+            selected={selected === item}
+            onSelect={() => {
+              setSelected(item);
+            }}
+          >
+            {item}
+          </ListboxPrimitiveItem>
+        ))}
+      </Stack>
+    </ListboxPrimitive>
+  )
+};
+
+render(
+  <DefaultExample />
+)
+`.trim();
+
+export const horizontalExample = `
+const HorizontalExample = () => {
+  const [items] = React.useState(['Item 1', 'Item 2', 'Item 3']);
+  const [selected, setSelected] = React.useState();
+  const listbox = useListboxPrimitiveState({orientation: 'horizontal'});
+  return (
+    <ListboxPrimitive {...listbox} aria-label="My listbox">
+      {items.map((item) => (
+        <ListboxPrimitiveItem
+          key={item}
+          {...listbox}
+          selected={selected === item}
+          onSelect={() => {
+            setSelected(item);
+          }}
+        >
+          {item}
+        </ListboxPrimitiveItem>
+      ))}
+    </ListboxPrimitive>
+  )
+};
+
+render(
+  <HorizontalExample />
+)
+`.trim();
+
+export const groupsExample = `
+const GroupsExample = () => {
+  const [items] = React.useState(['Item 1', 'Item 2', 'Item 3']);
+  const [selected, setSelected] = React.useState();
+  const renderItem = (item, index) => (
```

Comment:
issue: I'm generally not a fan of defining components within components. (`renderItem` can be renamed to `RenderItem` and used like `<RenderItem ... />`). Mind refactoring this out so it isn't re-initialized on re-renders please? People copy paste these examples and it could cause issues/be slow.